### PR TITLE
gh #62 l1 videoport testprofile improvement

### PR DESCRIPTION
**Pull request by thanushreevani**
updated videoport l1 file for test profile imporovement

### CODE REVIEW

File: src/test_l1_dsVideoPort.c [hunk 1]
```diff
@@ -750,13 +751,13 @@ void test_l1_dsVideoPort_positive_dsIsDisplaySurround(void) {
 	}
 
 	// Step 04: Repeat the check for audio surround support
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
 		status = dsIsDisplaySurround(handle[i], &isSurroundArray2[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 05: Compare the array values
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:759]: DisplaySurround is need to compare from configuration yaml file 
Please follow the same for other API's

**Review by hari22yuva — COMMENTED** [review]
Please compare or read the values from the yaml file 
remove the test cases if the header doesn't  that API's and remove from skeletons also 
while pushing the changes use gh #issuenumber as a commit message

File: src/test_l1_dsVideoPort.c [hunk 2]
```diff
@@ -73,6 +73,7 @@
 #include <stdlib.h>
 #include "dsVideoPort.h"
 #include "dsVideoPortSettings.h"
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done. 
for one variable invalid_kRes which is list of invalid resolutions added in the L1 file itself

File: src/test_l1_dsVideoPort.c [hunk 3]
```diff
@@ -591,35 +592,35 @@ void test_l1_dsVideoPort_positive_dsIsDisplayConnected(void) {
...
-	for (int i = 0; i < NUM_OF_PORTS ; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts ; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 03: Check if video port is connected
-	for (int i = 0; i < NUM_OF_PORTS ; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts ; i++) {
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:667]: Done

File: src/test_l1_dsVideoPort.c (hunk 1 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:822]: Done
for below values there is no correcponding param in yaml..?
1.dsHdcpStatus
2. dsColorDepthCapabilities

File: src/test_l1_dsVideoPort.c [hunk 4]
```diff
@@ -1144,22 +1145,23 @@ void test_l1_dsVideoPort_positive_dsEnableDTCP(void) {
...
-	for (int i = 0; i < NUM_OF_PORTS ; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &handle[i]);
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts ; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &handle[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 03: Enable DTCP
-	for (int i = 0; i < NUM_OF_PORTS ; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts ; i++) {
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: Done

File: src/test_l1_dsVideoPort.c [hunk 5]
```diff
@@ -1921,40 +1735,47 @@ void test_l1_dsVideoPort_positive_dsGetResolution(void) {
...
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
 		status = dsGetResolution(handle[i] , &resolutionArray2[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 05: Compare the array values
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		for (int j = 0; i < gDSVideoPortConfiguration[i].numSupportedResolutions; j++) {
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:1767]: not sure what value we get without set 
check for default values

File: src/test_l1_dsVideoPort.c [hunk 6]
```diff
@@ -5020,20 +4741,20 @@ void test_l1_dsVideoPort_positive_dsSetBackgroundColor(void) {
...
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &handle[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 03: Set background color for each port by looping through dsVideoBackgroundColor_t enum
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
 		for(dsVideoBackgroundColor_t color = dsVIDEO_BGCOLOR_BLUE; color < dsVIDEO_BGCOLOR_MAX ; color++){
 			status = dsSetBackgroundColor(handle[i], color);
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:4759]: check this implement as per header 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsVideoPort.h#L936
please check other API's also

File: src/test_l1_dsVideoPort.c [hunk 7]
```diff
@@ -5407,35 +5127,37 @@ void test_l1_dsVideoPort_positive_dsGetPreferredColorDepth(void) {
...
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 03: Get preferred color depth
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:5146]: please check 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsVideoPort.h#L1008

File: src/test_l1_dsVideoPort.c [hunk 8]
```diff
@@ -5618,13 +5338,13 @@ void test_l1_dsVideoPort_negative_dsSetPreferredColorDepth(void) {
...
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 05: Set unsupported color depth
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
 		status = dsSetPreferredColorDepth(handle[i], 0x60);
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:5348]: please check 
https://github.com/rdkcentral/rdk-halif-device_settings/blob/main/include/dsVideoPort.h#L1032

**Review by hari22yuva — COMMENTED** [review]
Please address the review comments

File: src/test_l1_dsVideoPort.c [hunk 9]
```diff
@@ -452,35 +509,35 @@ void test_l1_dsVideoPort_positive_dsIsVideoPortEnabled(void) {
...
+	bool isEnabledArray2[gDSvideoPort_NumberOfPorts];
 
 	// Step 01: Initialize video port system
 	status = dsVideoPortInit();
 	UT_ASSERT_EQUAL(status, dsERR_NONE);
 
 	// Step 02: Get the port handle for all supported video ports
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:473]: done

File: src/test_l1_dsVideoPort.c [hunk 10]
```diff
@@ -867,35 +932,36 @@ void test_l1_dsVideoPort_positive_dsGetSurroundMode(void) {
...
 		status = dsGetSurroundMode(handle[i], &surroundModeArray2[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 	}
 
 	// Step 05: Compare the array values
-	for (int i = 0; i < NUM_OF_PORTS  ; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts  ; i++) {
+		UT_ASSERT_EQUAL(surroundModeArray1[i],gDSVideoPortConfiguration[i].SurroundMode);
 		UT_ASSERT_EQUAL(surroundModeArray1[i], surroundModeArray2[i]);
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 11]
```diff
@@ -1273,11 +1221,12 @@ void test_l1_dsVideoPort_positive_dsEnableHDCP(void) {
...
 
 	// Step 02: Get the video port handle
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 
 		// Step 03: Enable HDCP
+		enableHDCP = gDSVideoPortConfiguration[i].hdcp_supported;
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 12]
```diff
@@ -1846,36 +1660,36 @@ void test_l1_dsVideoPort_negative_dsSetResolution(void) {
...
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
 		status = dsSetResolution(handle[i], NULL);
 		UT_ASSERT_EQUAL(status, dsERR_INVALID_PARAM);
 	}
 
 	// Step 06: Set resolution with invalid resolution parameters
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
 		for (int j = 0; j < (sizeof(invalid_kRes)/sizeof(invalid_kRes[0])); j++) {
 			status = dsSetResolution(handle[i], &invalid_kRes[j]);
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 5 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 6 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 7 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 8 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 1 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: removed the invalid_kRes. done

File: src/test_l1_dsVideoPort.c [hunk 13]
```diff
@@ -1007,35 +968,26 @@ void test_l1_dsVideoPort_positive_dsIsVideoPortActive(void) {
...
-	// Step 03: Check whether each video port is active
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+		// Step 03: Check whether each video port is active
 		status = dsIsVideoPortActive(handle[i], &isActiveArray1[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
-	}
-
-	// Step 04: Repeat the active status check
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
+		// Step 04: Repeat the active status check
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:983]: done

File: src/test_l1_dsVideoPort.c [hunk 14]
```diff
@@ -1273,14 +1104,16 @@ void test_l1_dsVideoPort_positive_dsEnableHDCP(void) {
...
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 
 		// Step 03: Enable HDCP
-		status = dsEnableHDCP(handle[i], enableHDCP, hdcpKey, keySize);
-		UT_ASSERT_EQUAL(status, dsERR_NONE);
-
+		if ((gSourceType == 1) && (gDSVideoPortConfiguration[i].hdcp_supported == true)) {
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 15]
```diff
@@ -1533,35 +1240,27 @@ void test_l1_dsVideoPort_positive_dsIsHDCPEnabled(void) {
...
-	}
-
-	// Step 05: Compare the array values
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		UT_ASSERT_EQUAL(isHDCPEnabledArray1[i], isHDCPEnabledArray2[i]);
+		// Step 05: Compare the array values with profile values
+		if ((gSourceType == 1)) {
+			UT_ASSERT_EQUAL(isHDCPEnabledArray1[i], gDSVideoPortConfiguration[i].hdcp_supported);
+		} else if ((gSourceType == 0)) {
+			UT_ASSERT_EQUAL(isHDCPEnabledArray1[i], true);
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:1257]: done

File: src/test_l1_dsVideoPort.c [hunk 16]
```diff
@@ -3251,12 +2739,9 @@ void test_l1_dsVideoPort_positive_dsSetForceDisable4KSupport(void) {
...
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		status = dsGetVideoPort(kPorts[i].id.type, kPorts[i].id.index, &(handle[i]));
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
-	}
-
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
 		// Step 03: Set force disable 4K support
 		status = dsSetForceDisable4KSupport(handle[i], disable4K);
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:2727]: done

File: src/test_l1_dsVideoPort.c [hunk 17]
```diff
@@ -3630,49 +3096,35 @@ void test_l1_dsVideoPort_negative_dsGetVideoEOTF(void) {
...
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
-	}
-
-	for (int i = 0; i < NUM_OF_PORTS; i++) { 
 		// Step 03: Retrieve the matrix coefficients
 		status = dsGetMatrixCoefficients(handle[i], &matrixCoefficientsArray1[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
+		// Step 04: Compare the array values
+		UT_ASSERT_EQUAL(matrixCoefficientsArray1[i], gDSVideoPortConfiguration[i].matrix_coefficients);
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:3100]: done

File: src/test_l1_dsVideoPort.c [hunk 18]
```diff
@@ -4213,33 +3605,23 @@ void test_l1_dsVideoPort_positive_dsGetCurrentOutputSettings(void) {
...
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &handle[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
 
 		// Step 03: Retrieve the current output settings
 		status = dsGetCurrentOutputSettings(handle[i], &hdrstandardarray1[i] ,&matrixcoefarray1[i], &colorspacearray1[i],\
 				&colordeptharray1[i], &quant_rangearray1[i]);
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
+		// Step 04: Compare the array values
+		UT_ASSERT_EQUAL(hdrstandardarray1[i], (dsHDRStandard_t)gDSVideoPortConfiguration[i].hdr_capabilities);
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 19]
```diff
@@ -4529,8 +3884,10 @@ void test_l1_dsVideoPort_positive_dsResetOutputToSDR(void) {
 	UT_ASSERT_EQUAL(status, dsERR_NONE);
 
 	// Step 02: Reset video output to SDR
-	status = dsResetOutputToSDR();
-	UT_ASSERT_EQUAL(status, dsERR_NONE);
+	if (gSourceType == 1) {
+		status = dsResetOutputToSDR();
+		UT_ASSERT_EQUAL(status, dsERR_NONE);
+	}
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:3879]: done

File: src/test_l1_dsVideoPort.c [hunk 20]
```diff
@@ -4883,36 +4223,29 @@ void test_l1_dsVideoPort_positive_dsGetIgnoreEDIDStatus(void) {
...
+		if (gSourceType == 1) {
+			// Step 03: Get IgnoreEDID status
+			status = dsGetIgnoreEDIDStatus(handle[i], &ignoreEDIDStatus1[i]);
+			UT_ASSERT_EQUAL(status, dsERR_NONE);
+			// Step 04: Repeat getting IgnoreEDID status
+			status = dsGetIgnoreEDIDStatus(handle[i], &ignoreEDIDStatus2[i]);
+			UT_ASSERT_EQUAL(status, dsERR_NONE);
+			// Step 05: Compare the array values and make sure they are equal
+			UT_ASSERT_EQUAL(ignoreEDIDStatus1[i], ignoreEDIDStatus2[i]);
+		}
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:4232]: done

File: src/test_l1_dsVideoPort.c [hunk 21]
```diff
@@ -5020,23 +4356,22 @@ void test_l1_dsVideoPort_positive_dsSetBackgroundColor(void) {
...
-			UT_ASSERT_EQUAL(status, dsERR_NONE);
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &handle[i]);
+		UT_ASSERT_EQUAL(status, dsERR_NONE);
+			// Step 03: Set background color for each port by looping through dsVideoBackgroundColor_t enum
+		if (gSourceType == 1) {
+			for(dsVideoBackgroundColor_t color = dsVIDEO_BGCOLOR_BLUE; color < dsVIDEO_BGCOLOR_MAX ; color++){
+				status = dsSetBackgroundColor(handle[i], color);
+				UT_ASSERT_EQUAL(status, dsERR_NONE);
+			}
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:4354]: done

File: src/test_l1_dsVideoPort.c [hunk 22]
```diff
@@ -5202,38 +4537,42 @@ void test_l1_dsVideoPort_negative_dsSetForceHDRMode(void) {
...
 	}
 
-	// Step 06: Terminate the video port system
+	if (gSourceType == 0) {
+		// Step 06: Set HDR mode for unsupported device
+		status = dsSetForceHDRMode(handle[0], gDSVideoPortConfiguration[0].hdr_capabilities);
+		UT_ASSERT_EQUAL(status, dsERR_OPERATION_NOT_SUPPORTED);
+	}
+
+	// Step 07: Terminate the video port system
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 23]
```diff
@@ -5268,36 +4607,25 @@ void test_l1_dsVideoPort_positive_dsColorDepthCapabilities(void) {
...
-	// Step 05: Compare array values
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		UT_ASSERT_EQUAL(colorDepthCapability1[i], colorDepthCapability2[i]);
+		if (gSourceType == 1) {
+			// Step 03: Get color depth capabilities
+			status = dsColorDepthCapabilities(handle[i], &colorDepthCapability1[i]);
+			UT_ASSERT_EQUAL(status, dsERR_NONE);
+			// Step 05: Compare array values with values from profile file
+			UT_ASSERT_EQUAL(colorDepthCapability1[i], gDSvideoPort_color_depth);
+		}
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 24]
```diff
@@ -5397,46 +4727,36 @@ void test_l1_dsVideoPort_negative_dsColorDepthCapabilities(void) {
...
-	// Step 05: Compare array values
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		UT_ASSERT_EQUAL(colorDepth1[i], colorDepth2[i]);
+	if (gSourceType == 1) {
+		for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+			status = dsGetPreferredColorDepth(handle[i], &colorDepth1[i]);
+			UT_ASSERT_EQUAL(status, dsERR_NONE);
+			dsDisplayColorDepth_t colorDepth = getColorDepth();
+			UT_ASSERT_EQUAL(colorDepth1[i], colorDepth);
+		}
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 25]
```diff
@@ -5536,29 +4858,27 @@ void test_l1_dsVideoPort_negative_dsGetPreferredColorDepth(void) {
...
+	for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+		status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle[i]));
 		UT_ASSERT_EQUAL(status, dsERR_NONE);
-	}
-
-	// Step 03: Set preferred color depth
-	for (int i = 0; i < NUM_OF_PORTS; i++) {
-		for (dsDisplayColorDepth_t colorDepth = dsDISPLAY_COLORDEPTH_UNKNOWN; colorDepth <= dsDISPLAY_COLORDEPTH_AUTO; colorDepth++ ) {
+		// Step 03: Set preferred color depth
+		if (gSourceType == 1) {
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c:4843]: done

File: src/test_l1_dsVideoPort.c [hunk 26]
```diff
@@ -107,6 +107,20 @@ static bool extendedEnumsSupported=false; //Default to not supported
    }\
 }
 
+dsDisplayColorDepth_t getColorDepth(void){
+	dsDisplayColorDepth_t colorDepth;
+	if (gDSvideoPort_color_depth == 8) {
```
[the remark this answers is not in the document]
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 27]
```diff
@@ -315,42 +329,42 @@ void test_l1_dsVideoPort_negative_dsVideoPortTerm(void) {
...
+
+    dsError_t status;
+
+    intptr_t handle[gDSvideoPort_NumberOfPorts];
+    intptr_t lastHandle;
+    intptr_t newHandle;
+
+    // Step 01: Initialize video port system
+    status = dsVideoPortInit();
+    UT_ASSERT_EQUAL(status, dsERR_NONE);
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:343]: use  UT_ASSERT_EQUAL_FATAL() for dsVideoPortInit(), dsVideoPortTerm() 
same way like L2 testcases

File: src/test_l1_dsVideoPort.c [hunk 28]
```diff
@@ -1002,48 +959,39 @@ void test_l1_dsVideoPort_negative_dsGetSurroundMode(void) {
...
-	status = dsVideoPortTerm();
-	UT_ASSERT_EQUAL(status, dsERR_NONE);
-
-	UT_LOG("\n Out %s\n", __FUNCTION__); 
+    gTestID = 15;
+    UT_LOG("\n In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status;
+
+    intptr_t handle[gDSvideoPort_NumberOfPorts];
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:967]: do we required the array of handles?
please check other places also

File: src/test_l1_dsVideoPort.c [hunk 29]
```diff
@@ -1002,48 +959,39 @@ void test_l1_dsVideoPort_negative_dsGetSurroundMode(void) {
...
+
+    bool isActiveArray1[gDSvideoPort_NumberOfPorts];
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:969]: in single loop only we are comparing so may not required array
other places also if not required remove array and use single variable

**Review by hari22yuva — COMMENTED** [review]
partial review 
please check these comments

File: src/test_l1_dsVideoPort.c (hunk 27 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 28 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c [hunk 30]
```diff
@@ -1264,53 +1215,49 @@ void test_l1_dsVideoPort_negative_dsEnableHDCP(void) {
...
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        // Step 03: Check if HDCP is enabled
+        status = dsIsHDCPEnabled(handle, &isHDCPEnabled1);
+        // Step 04: Compare the array values with profile values
+        if ((gSourceType == 1)) {
+            if (gDSVideoPortConfiguration[i].hdcp_supported == false) {
+                UT_ASSERT_EQUAL(status, dsERR_OPERATION_NOT_SUPPORTED);
+            } else {
+                UT_ASSERT_EQUAL(status, dsERR_NONE);
+                 UT_ASSERT_EQUAL(isHDCPEnabled1, gDSVideoPortConfiguration[i].hdcp_supported);
```
Comment (drissya-k) [inline src/test_l1_dsVideoPort.c:1248]: An extra space

File: src/test_l1_dsVideoPort.c [hunk 31]
```diff
@@ -1264,53 +1215,49 @@ void test_l1_dsVideoPort_negative_dsEnableHDCP(void) {
...
+    UT_ASSERT_EQUAL_FATAL(status, dsERR_NONE);
+
+    // Step 02: Get the video port handle
+    for (int i = 0; i < gDSvideoPort_NumberOfPorts; i++) {
+        status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle));
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        // Step 03: Check if HDCP is enabled
+        status = dsIsHDCPEnabled(handle, &isHDCPEnabled1);
+        // Step 04: Compare the array values with profile values
+        if ((gSourceType == 1)) {
```
Comment (drissya-k) [inline src/test_l1_dsVideoPort.c:1243]: double parentheses

File: src/test_l1_dsVideoPort.c (hunk 30 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

File: src/test_l1_dsVideoPort.c (hunk 31 above)
Comment (thanushreevani) [reply src/test_l1_dsVideoPort.c]: done

**Review by drissya-k — APPROVED** [review]

File: src/test_l1_dsVideoPort.c [hunk 32]
```diff
@@ -581,53 +572,45 @@ void test_l1_dsVideoPort_negative_dsIsVideoPortEnabled(void) {
...
+    for (int i = 0; i < gDSvideoPort_NumberOfPorts ; i++) {
+        status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle));
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        // Step 03: Check if video port is connected
+        status = dsIsDisplayConnected(handle, &isConnected1);
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        // Step 04: Compare the array values
+        bool isConnected;
+        if (gSourceType == 0) {
+            isConnected = true;
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:602]: we can directly do the assert UT_ASSERT_EQUAL(isConnected1, TRUE);
no need of extra variable.
you can take this in clean up activity.

File: src/test_l1_dsVideoPort.c [hunk 33]
```diff
@@ -2980,53 +2809,44 @@ void test_l1_dsVideoPort_negative_dsSetForceDisable4KSupport(void) {
...
+        status = dsGetVideoPort(gDSVideoPortConfiguration[i].typeid, gDSVideoPortConfiguration[i].index, &(handle));
+        UT_ASSERT_EQUAL(status, dsERR_NONE);
+        // Step 03: Retrieve whether 4K support is disabled
+        status = dsGetForceDisable4KSupport(handle, &disable4K1);
+        UT_ASSERT_EQUAL(status, dsERR_OPERATION_NOT_SUPPORTED);
+        // Step 04: Repeat the retrieval
+        status = dsGetForceDisable4KSupport(handle, &disable4K2);
+        UT_ASSERT_EQUAL(status, dsERR_OPERATION_NOT_SUPPORTED);
+        // Step 05: Compare the array values
+        UT_ASSERT_EQUAL(disable4K1, disable4K2);
```
Comment (hari22yuva) [inline src/test_l1_dsVideoPort.c:2842]: i think we may need to compare 
this also we take care in code clan up activity

**Review by hari22yuva — APPROVED** [review]
Approved

**Pull request merged**